components/Overlay/PositionMarketOverlay/index.tsx

```diff
@@ -1,32 +1,75 @@
 import React, { FC, useEffect, useState } from 'react';
 import styled from 'styled-components';
 import Overlay from '@components/Overlay';
-import { Logo } from '@components/General';
+import { Button, Logo } from '@components/General';
 import { toApproxCurrency } from '@libs/utils';
 import MarketChange from '@components/General/MarketChange';
 import { LabelledTracers } from '@libs/types/TracerTypes';
 import { PortfolioDropdown } from '@components/Portfolio';
 import Tracer from '@libs/Tracer';
+import Link from 'next/link';
 
 interface POProps {
     tracers: LabelledTracers;
 }
 const PositionMarketOverlay: FC<POProps> = ({ tracers }: POProps) => {
     const [currentMarket, setCurrentMarket] = useState('');
     const [marketKeyMap, setMarketKeyMap] = useState<Record<string, string>>({});
+    const [count, setCount] = useState(0);
+
+    const [show, setShow] = useState(NaN);
+    const updateTimer = React.useRef(null);
+
+    function setUpdate() {
+        setShow(0);
+        // @ts-ignore
+        updateTimer.current = setTimeout(() => {
+            setShow(1);
+            updateTimer.current = null;
+        }, 100);
+    }
+
+    useEffect(() => {
+        if (!updateTimer.current) {
+            setUpdate();
+        }
+    }, [count]);
+
+    useEffect(() => {
+        return () => {
+            if (updateTimer.current) {
+                // @ts-ignore
+                clearTimeout(updateTimer.current);
+            }
+        };
+    }, []);
+
+    useEffect(() => {
+        const interval = setInterval(() => {
+            // Increment count every 6 seconds
+            if (count === Object.keys(marketKeyMap).length - 1) {
+                setCount(0);
+            } else {
+                setCount(count + 1);
+            }
+        }, 6000);
+        setCurrentMarket(Object.keys(marketKeyMap)[count]);
+        return () => {
+            clearInterval(interval);
+        };
+    }, [count]);
```

Comment:
setShow is confusing here its a number but as far as I can see its only getting set to 0 or 1 so it should be a boolean.
Also its calling useEffect on count which creates a new interval. I don't think the interval creation should be in here. I think it could be cleaned up a little try and reduce it.